opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -296,277 +196,151 @@ class Restart : private boost::noncopyable
-    }
-
-    // no destructor in this class as ERT takes care of freeing the
-    // handle as part of freeing the solution handle!
-
-    ecl_sum_tstep_type *ertHandle() const
-    { return ertHandle_; };
-
-private:
-    ecl_sum_tstep_type *ertHandle_;
-};
-
-
 /**
  * Initialization file which contains static properties (such as
  * porosity and permeability) for the simulation field.
  */
-class Init : private boost::noncopyable
-{
+class Init {
 public:
-    Init(const std::string& outputDir,
-         const std::string& baseName,
-         int writeStepIdx,
-         IOConfigConstPtr ioConfig) : egridFileName_(outputDir,
-                                                     baseName,
-                                                     ECL_EGRID_FILE,
-                                                     writeStepIdx,
-                                                     ioConfig->getFMTOUT())
-    {
-        bool formatted = ioConfig->getFMTOUT();
-
-        FileName initFileName(outputDir,
-                              baseName,
-                              ECL_INIT_FILE,
-                              writeStepIdx,
-                              formatted);
-
-        ertHandle_ = fortio_open_writer(initFileName.ertHandle(),
-                                        formatted,
-                                        ECL_ENDIAN_FLIP);
-    }
+    Init( const std::string& outputDir,
+          const std::string& baseName,
+          int writeStepIdx,
+          const IOConfig& ioConfig ) :
+        egrid( outputDir,
+               baseName,
+               ECL_EGRID_FILE,
+               writeStepIdx,
+               ioConfig.getFMTOUT() ),
+        init( fortio_open_writer(
+                    FileName( outputDir,
+                        baseName,
+                        ECL_INIT_FILE,
+                        writeStepIdx,
+                        ioConfig.getFMTOUT() ).ertHandle(),
+                    ioConfig.getFMTOUT(),
+                    ECL_ENDIAN_FLIP ) )
 
-    ~Init()
-    { fortio_fclose(ertHandle_); }
+    {}
 
-    void writeHeader(int numCells,
-                     const int* compressedToCartesianCellIdx,
-                     const SimulatorTimerInterface& timer,
-                     Opm::EclipseStateConstPtr eclipseState,
-                     const PhaseUsage uses,
-                     const NNC& nnc = NNC())
+    void writeHeader( int numCells,
+                      const int* compressedToCartesianCellIdx,
+                      time_t current_posix_time,
+                      const EclipseState& es,
+                      int ert_phase_mask,
+                      const NNC& nnc = NNC() )
     {
-        auto dataField = eclipseState->get3DProperties().getDoubleGridProperty("PORO").getData();
-        restrictAndReorderToActiveCells(dataField, numCells, compressedToCartesianCellIdx);
+        auto dataField = es.get3DProperties().getDoubleGridProperty("PORO").getData();
```

Comment:
This is for the INIT file though in which case we're interested in writing whatever is in the Deck (no?), rather than what the simulator sets up internally later.

Correct me if I'm wrong in this, but I'd assume that meant outputting whatever is provided by the Deck (and so EclipseState).